File: src/main/java/engine/core/state/GameState.java
package engine.core.state;

import com.jogamp.opengl.GLAutoDrawable;

import java.awt.event.KeyEvent;

public interface GameState {
    void init(GLAutoDrawable glAutoDrawable);
    void dispose(GLAutoDrawable glAutoDrawable);
    void display(GLAutoDrawable glAutoDrawable);
    void reshape(GLAutoDrawable glAutoDrawable, int i, int i1, int i2, int i3);
    void update(float deltaTime);

    //void actionPerformed(ActionEvent e, int deltaTime);
    void keyTyped(KeyEvent e);
    void keyPressed(KeyEvent e);
    void keyReleased(KeyEvent e);
}

File: src/main/java/engine/core/state/GameState.kt
package engine.core.state

import com.jogamp.opengl.GLAutoDrawable

import java.awt.event.KeyEvent

interface GameState {
    fun init(glAutoDrawable: GLAutoDrawable)
    fun dispose(glAutoDrawable: GLAutoDrawable)
    fun display(glAutoDrawable: GLAutoDrawable)
    fun reshape(glAutoDrawable: GLAutoDrawable, i: Int, i1: Int, i2: Int, i3: Int)
    fun update(deltaTime: Float)

    //void actionPerformed(ActionEvent e, int deltaTime);
    fun keyTyped(e: KeyEvent)

    fun keyPressed(e: KeyEvent)
    fun keyReleased(e: KeyEvent)
}
